doc/mgmt/SONiC_OC_Interface_Enhancements_HLD.md

```diff
@@ -0,0 +1,373 @@
+5. When the configuration changes from admin down to up the operational status will change to operationally up.   
+
+### 1.3.2 Container
+* **swss container** : 
+  * VlanMgr and Orchagent changes to set oper status based on admin status.
+* **management-framework** :
+  * CLI XML & Jinja-template changes to support CLI-style configurations and relevant show commands.
+  
+### 1.3.3 SAI Overview
+N.A
+
+# 2 Functionality
+As Described in section 1.2
+
+# 3 Design
+## 3.1 Overview
+As described in section 1.3
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+**VLAN table**
+* No change.
+
+**LOOPBACK table**
+* Producer: Mangement framework
+* Consumer: VlanMgr
+* Description: Update existing table to store 'admin_status' configuration.
+* Schema:
+```
+;Existing table
+;defines Loopback. Store admin state configuration 
+;
+;Status: stable
+
+key = LOOPBACK_TABLE|LOOPBACK_NAME ;
+admin_status = "up"/"down" ; admin status
+```
+
+### 3.2.2 APP DB
+**VLAN_TABLE** 
+* No change.
+
+**LOOPBACK_TABLE** 
+* No change.
+
+### 3.2.3 STATE DB
+N/A
+### 3.2.4 ASIC DB
+N/A
+### 3.2.5 COUNTER DB
+N/A
+### 3.2.6 ERROR DB
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+As in sec 1.3
+### 3.3.2 Other Processes 
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+
+#### 3.6.1.1 SONiC Yang
+#### VLAN
+1. sonic-vlan.yang
+* Containers VLAN and VLAN_TABLE already have leaf admin status as below:
+```
+leaf admin_status {
+    type scommon:admin-status;
+}
+```
+#### LOOPBACK
+1. sonic-loopback.yang
+* Update containers LOOPBACK and LOOPBACK_TABLE with leaf admin status as below:
+```
+leaf admin_status {
+    type scommon:admin-status;
+}
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+
+#### VLAN
+```
+sonic(conf-if-Vlan100)# [no] shutdown
+sonic(conf-if-range-vl**)# [no] shutdown
+```
+#### LOOPBACK
+```
+sonic(conf-if-lo10)# [no] shutdown
+```
+
+#### 3.6.2.2 Show Commands
+
+#### VLAN
+* "show vlan" displays the Vlan Interface’s operational status.
+```
+sonic# show Vlan
+Q: A - Access (Untagged), T - Tagged
+NUM        Status      Q Ports        AutoState
+100        Inactive    T  Ethernet0   Enable
+                       T  Ethernet1
+101        Active      T  Ethernet1   Disable
+```
+* "show interface vlan" displays both the Admin & Oper status
+```
+sonic(config)# do show interface Vlan 1
+Vlan1 is up, line protocol is down
+Mode of IPV4 address assignment: not-set
+Mode of IPV6 address assignment: not-set
+Interface IPv6 oper status: Disabled
+IP MTU 9100 bytes
+sonic(config)#
+```
+* "show configuration" to display "shutdown" configuration.  
+```
+sonic(config)# do show running-configuration interface Vlan 1
+!
+interface Vlan1
+ shutdown
+```
+
+#### LOOPBACK
+* "show interface Loopback" displays Oper status
+```
+sonic#show interface Loopback 10
+Loopback10 is up
```

Comment:
Will this not be enhanced to display the admin status of LO intf?

Reply:
Based on discussed with team, it would be useful to have both the admin and oper status in the show command.
Will update the HLD.